diff --git a/geolocation.install b/geolocation.install
new file mode 100644
index 0000000..c798a9d
--- /dev/null
+++ b/geolocation.install
@@ -0,0 +1,18 @@
+<?php
+
+/**
+ * @file
+ * Contains update for the geolocation module.
+ */
+
+/**
+ * @param $sandbox
+ *
+ * Implements hook_update_N().
+ * Set a empty default value for geolocation.google_map_api_key.
+ */
+function geolocation_update_8001($sandbox) {
+  $config = Drupal::configFactory()->getEditable('geolocation.settings');
+  $config->set('geolocation.google_map_api_key', '');
+  $config->save();
+}
diff --git a/geolocation.links.menu.yml b/geolocation.links.menu.yml
new file mode 100644
index 0000000..ceb8a8f
--- /dev/null
+++ b/geolocation.links.menu.yml
@@ -0,0 +1,12 @@
+# Define default links for this module.
+geolocation.admin_config_geolocation:
+  route_name: geolocation.admin_config_geolocation
+  title: 'Geolocation settings'
+  parent: system.admin_config‚
+  description: 'Configure geolocation settings.'
+
+geolocation.google_map_api_key:
+  route_name: geolocation.google_map_api_key
+  title: 'Geolocaction - Google Map API key'
+  parent: geolocation.admin_config_geolocation
+  description: 'A simple form to add a Google Map API key'
\ No newline at end of file
diff --git a/geolocation.permissions.yml b/geolocation.permissions.yml
new file mode 100644
index 0000000..19a131f
--- /dev/null
+++ b/geolocation.permissions.yml
@@ -0,0 +1,2 @@
+configure geolocation:
+  title: 'Configure Geolocation'
\ No newline at end of file
diff --git a/geolocation.routing.yml b/geolocation.routing.yml
new file mode 100644
index 0000000..261fb74
--- /dev/null
+++ b/geolocation.routing.yml
@@ -0,0 +1,15 @@
+geolocation.admin_config_geolocation:
+  path: '/admin/config/geolocation'
+  defaults:
+    _controller: '\Drupal\system\Controller\SystemController::systemAdminMenuBlockPage'
+    _title: 'Geolocation settings'
+  requirements:
+    _permission: 'configure geolocation'
+
+geolocation.google_map_api_key:
+  path: '/admin/config/geolocation/google-map-api'
+  defaults:
+    _form:  '\Drupal\geolocation\Form\GeolocationGoogleMapAPIkey'
+    _title: 'Geolocaction - Google Maps API key'
+  requirements:
+    _permission: 'configure geolocation:'
\ No newline at end of file
diff --git a/js/geolocation.js b/js/geolocation.js
index a04e384..0978ec5 100644
--- a/js/geolocation.js
+++ b/js/geolocation.js
@@ -58,21 +58,43 @@
   };
 
   /**
+   * ToDo: Find a better way to get the key!
+   */
+  geolocation.getGoogleAPIkey = function() {
+    var maps = settings.geolocation.maps,
+      firstMap = maps[Object.keys(maps)[0]];
+
+    return firstMap.google_api_key;
+  };
+
+  /**
    * Load google maps and set a callback to run when it's ready.
    *
    * @param callback
    */
   geolocation.load_google = function(callback) {
+    // Get google_api_key.
+    var google_api_key = geolocation.getGoogleAPIkey();
     // Add the callback.
     geolocation.add_callback(callback);
     // Check for google maps.
     if (!geolocation.maps_api_loading && (typeof google == 'undefined' || typeof google.maps == 'undefined')) {
       geolocation.maps_api_loading = true;
       // google maps isn't loaded so lazy load google maps.
-      $.getScript("//maps.googleapis.com/maps/api/js?v=3.exp&sensor=false&callback=Drupal.geolocation.google_callback")
-        .done(function() {
-          geolocation.maps_api_loading = false;
-        });
+      // If a Google API key isset, use it.
+      if(google_api_key != '') {
+        $.getScript("//maps.googleapis.com/maps/api/js?key="+google_api_key+"&v=3.exp&sensor=false&callback=Drupal.geolocation.google_callback")
+          .done(function() {
+            geolocation.maps_api_loading = false;
+          });
+      }
+      else {
+        $.getScript("//maps.googleapis.com/maps/api/js?v=3.exp&sensor=false&callback=Drupal.geolocation.google_callback")
+          .done(function() {
+            geolocation.maps_api_loading = false;
+          });
+      }
+
     } else {
       // Google maps loaded. Run callback.
       geolocation.google_callback();
diff --git a/src/Plugin/Field/FieldFormatter/GeolocationGoogleMapFormatter.php b/src/Plugin/Field/FieldFormatter/GeolocationGoogleMapFormatter.php
index 4725ea8..94cecf6 100644
--- a/src/Plugin/Field/FieldFormatter/GeolocationGoogleMapFormatter.php
+++ b/src/Plugin/Field/FieldFormatter/GeolocationGoogleMapFormatter.php
@@ -172,6 +172,9 @@ class GeolocationGoogleMapFormatter extends FormatterBase {
 
       $uniqueue_id = uniqid("map-canvas-");
 
+      // Add Google API key to js.
+      $config = \Drupal::config('geolocation.settings');
+
       $elements[$delta] = [
         '#type' => 'markup',
         '#markup' => '<div id="' . $uniqueue_id . '" class="geolocation-google-map"></div>',
@@ -185,6 +188,7 @@ class GeolocationGoogleMapFormatter extends FormatterBase {
                   'lat' => (float)$item->lat,
                   'lng' => (float)$item->lng,
                   'settings' => $field_settings,
+                  'google_api_key' => $config->get('geolocation.google_map_api_key', ''),
                 ],
               ],
             ],
